source/DSCResources/DSC_CMMaintenanceWindows/DSC_CMMaintenanceWindows.psm1

```diff
@@ -0,0 +1,520 @@
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $CollectionName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $Name,
+
+        [Parameter()]
+        [ValidateSet('Any','SoftwareUpdatesOnly','TaskSequencesOnly')]
+        [String]
+        $ServiceWindowsType = 'Any',
+
+        [Parameter()]
+        [String]
+        $Start,
+
+        [Parameter()]
+        [ValidateSet('MonthlyByDay','MonthlyByWeek','Weekly','Days','None')]
+        [String]
+        $ScheduleType,
+
+        [Parameter()]
+        [UInt32]
+        $RecurInterval,
+
+        [Parameter()]
+        [ValidateSet('First','Second','Third','Fourth','Last')]
+        [String]
+        $MonthlyWeekOrder,
+
+        [Parameter()]
+        [ValidateSet('Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday')]
+        [String]
+        $DayOfWeek,
+
+        [Parameter()]
+        [ValidateRange(0,31)]
+        [UInt32]
+        $DayOfMonth,
+
+        [Parameter()]
+        [ValidateRange(1,23)]
+        [UInt32]
+        $HourDuration,
+
+        [Parameter()]
+        [ValidateRange(5,59)]
+        [UInt32]
+        $MinuteDuration,
+
+        [Parameter()]
+        [Boolean]
+        $IsEnabled = $true,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -CollectionName $CollectionName -Name $Name
+    $result = $true
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($PSBoundParameters.ContainsKey('HourDuration') -and $PSBoundParameters.ContainsKey('MinuteDuration'))
+        {
+            Write-Warning -Message $script:localizedData.MixedDuration
+        }
+
+        if ($state.CollectionStatus -eq 'Absent')
+        {
+            Write-Warning -Message ($script:localizedData.MissingCollection -f $CollectionName)
+            $result = $false
+        }
+        elseif ($state.Ensure -eq 'Absent')
+        {
+            if ((-not $PSBoundParameters.ContainsKey('ScheduleType')) -or
+                (($PSBoundParameters.ContainsKey('ScheduleType')) -and
+                (-not $PSBoundParameters.ContainsKey('MinuteDuration') -and
+                -not $PSBoundParameters.ContainsKey('HourDuration'))))
+            {
+                Write-Warning -Message ($script:localizedData.MissingWindowParam -f $Name)
+            }
+
+            Write-Warning -Message ($script:localizedData.MissingWindow -f $Name)
+            $result = $false
+        }
+        else
+        {
+            $valuesToCheck = @('CollectionName','Name','IsEnabled','Ensure','ServiceWindowsType')
+
+            $testParams = @{
+                CurrentValues = $state
+                DesiredValues = $PSBoundParameters
```

Comment:
$PSBoundParameters doesn't pull in default values, so we need to add IsEnabled as $true above if $PSBoundParameters doesn't contain IsEnabled.

if (-not $PSBoundParameters.ContainsKey('IsEnabled')) 
            {
                $PSBoundParameters.Add('IsEnabled',$true)
            }